prober/tcp.go

```diff
@@ -46,9 +46,22 @@ func dialTCP(ctx context.Context, target string, module config.Module, registry
 
 	if ip.IP.To4() == nil {
 		dialProtocol = "tcp6"
+		sourceNetwork = "ip6"
 	} else {
 		dialProtocol = "tcp4"
+		sourceNetwork = "ip4"
 	}
+
+	if len(module.SourceAddress) > 0 {
+		saddr, err := net.ResolveIPAddr(sourceNetwork, module.SourceAddress)
```

Comment:
Why do you resolve this here and not elsewhere? I don't think there's any need to resolve it.

Reply:
When module.SourceAddress is an IP-Address already no resolving will happen. In that case net.ResolveIPAddr will just convert to an net.IPAddr. I did not want to prevent names being used as module.SourceAddress so resolving would be necessary.
As to "why here not elsewhere". dialTcp() is the place where the target is resolved as well (chooseProtocol) and resolving in the dial function seems to be aligned with golang dialers in general. Which place do you think would be better suited for resolving?

Reply:
I'm saying that in some probes you support dns names and in others you don't. I'm suggesting not supporting it anywhere.

Reply:
Ahh! Actually resolving is in for DNS, TCP and HTTP. Only ICMP (initial PR) is using net.ParseIp.
The resolving also solves the question of using IPv4 or IPv6 which also depends on the target (You need a local v6 address for a v6 target but a v4 address for a v4 taget).
A DNS name could provide v4 and v6 address. Going with IP-addresses only in configuration would require
us to provide both (v4/v6). That is why I concluded that using a resolve-step to be a good idea.
When only an IP address is specified in config there is no resolving happening but only IP parsing.

Reply:
Relying on DNS to handle v4 vs v6 is a bit odd to me. I'd expect the user to provide only the address type that they need. Users requiring this are going to have a good idea of how networking works.